packages/@aws-cdk/aws-appmesh/lib/virtual-gateway.ts

```diff
@@ -173,15 +174,19 @@ export class VirtualGateway extends VirtualGatewayBase {
     }
 
     const accessLogging = props.accessLog?.bind(this);
+    const tlsClientPolicy = props.backendDefaults?.tlsClientPolicy;
 
     const node = new CfnVirtualGateway(this, 'Resource', {
       virtualGatewayName: this.physicalName,
       meshName: this.mesh.meshName,
       spec: {
         listeners: this.listeners.map(listener => listener.listener),
-        backendDefaults: props.backendDefaults !== undefined
+        backendDefaults: props.backendDefaults
           ? {
-            clientPolicy: props.backendDefaults?.clientPolicy?.bind(this).clientPolicy,
+            clientPolicy: {
+              tls: tlsClientPolicy ? renderTlsPolicy(this, tlsClientPolicy,
+                (config) => config.virtualGatewayClientTlsValidationTrust) : undefined,
```

Comment:
Let's make `renderTlsPolicy` handle the undefined case, we'll be able to say here then:

```suggestion
              tls: renderTlsPolicy(this, tlsClientPolicy, (config) => config.virtualGatewayClientTlsValidationTrust),
```

Much simpler.

Reply:
I tried this path and realized there is a tradeoff to make. 
Currently, `renderTlsPolicy` returns type = `CfnVirtualNode.ClientPolicyTlsProperty`. This type does not allow `undefined` to be assigned. 
In order to make `renderTlsPolicy` handle the undefined case, it must return type = any. I tried to avoid using `any` type since you mentioned [here](https://github.com/aws/aws-cdk/pull/14782#discussion_r636325552) types did not look good.

Reply:
Simple solution: make `renderTlsPolicy()` return `CfnVirtualNode.ClientPolicyTlsProperty | undefined`. Problem solved 🙂.

Reply:
Of course! I somehow overlooked that union type can be used for the return type as well.